pkg/server/plugin/handler.go

```diff
@@ -56,12 +56,22 @@ func NewPluginHandler(info pluginHandlerInfo, ppreg router.PreprocessorRegistry,
 func (h *Handler) Setup() {
 	if h.UserRequired {
 		h.preprocessors = h.PreprocessorList.GetByNames(
-			"plugin", "authenticator", "dbconn", "inject_user", "require_user")
+			"authenticator",
+			"dbconn",
+			"inject_user",
+			"require_user",
+			"plugin_ready",
+		)
 	} else if h.AccessKeyRequired {
 		h.preprocessors = h.PreprocessorList.GetByNames(
-			"plugin", "authenticator")
+			"authenticator",
```

Comment:
Ah I see now, either `accesskey` or `authenticator` is required. So `authenticator` fits the bill here.